hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -97,24 +108,63 @@ def get_neighbors(array, i, j, n=1):
     return array[i_start:i_stop, j_start:j_stop]
 
 
-def ref_point_with_max_cc(data_cc: np.array, window_size=1):
-    '''
-    shift determine the window size, n=1 9-pixel window, n=2, 25-pixel window, etc.
-    '''
+def find_ref_point_from_candidates(data_cc: np.ndarray, indices: (np.array, np.array),
+                                   window_size, start_idx, pick_num):
+
+    rows = indices[0][start_idx:start_idx + pick_num]
 
-    data_cc_max = data_cc[data_cc < 1.0].max()
-    idx = np.where(data_cc == data_cc_max)
+    cols = indices[1][start_idx:start_idx + pick_num]
 
-    rows, cols = idx[0], idx[1]
     num = len(rows)
     tots = np.zeros(num, dtype=float)
 
     for k in range(num):
-        tots[k] = get_neighbors(data_cc, rows[k], cols[k], window_size).sum()
+        neighbors = get_neighbors(data_cc, rows[k], cols[k], window_size)
+        if (neighbors == 0.0).any() or neighbors.size < (2*window_size+1)**2:
+            tots[k] = 0.0
+        else:
+            tots[k] = neighbors.sum()
```

Comment:
Since `tots` is already filled with zeros, could you just simplify like this?
```suggestion
        if not ((neighbors == 0.0).any() or neighbors.size < (2*window_size+1)**2):
            tots[k] = neighbors.sum()
```